### PR TITLE
osd/OSD.h: add override to virtual functions

### PR DESCRIPTION
**Pull request by stiopaa1**
Signed-off-by: Michal Jarzabek <stiopa@gmail.com>

### CODE REVIEW

File: src/osd/OSD.h [hunk 1]
```diff
@@ -1934,15 +1934,15 @@ class OSD : public Dispatcher,
 	}
       }
     }
-    bool _enqueue(PG *pg) {
+    bool _enqueue(PG *pg) override {
```
Comment (tchaikov) [inline src/osd/OSD.h]: can we add `noexcept` for better performance?

Comment (stiopaa1) [inline src/osd/OSD.h]: @tchaikov 
I don't have much experience with noexcept, so please correct me if i'm wrong. 
My understanding is that we use noexcept for functions that are guaranteed not to throw exceptions, but here the push_back might throw bad_alloc exception?

Comment (tchaikov) [inline src/osd/OSD.h]: > I don't have much experience with noexcept

neither do i =)

i think the actual questions are:

1. we need to mark the overrided method in the parent class with `noexcept`, if we want to have it in the child class.
1. shall we `terminate()` at seeing an exception we do not handle?

**Comment by tchaikov** [comment]
lgtm

**Pull request merged**